autoload/neoformat.vim

```diff
@@ -1,13 +1,18 @@
 function! neoformat#Neoformat(bang, user_input, start_line, end_line) abort
-    let search = @/
     let view = winsaveview()
+    let search = @/
     let original_filetype = &filetype
 
     call s:neoformat(a:bang, a:user_input, a:start_line, a:end_line)
 
+    " Setting &filetype might destroy existing folds, so only do that
+    " if the filetype got changed (which can only be possible when
```

Comment:
why can the filetype only be changed with the `bang`?

Reply:
Because the only place where we end up changing `&filetype` is inside `s:neoformat()` ([link](https://github.com/iamFIREcracker/neoformat/blob/82816be0bcf3d9cf15b2f742802369d1492e8dcd/autoload/neoformat.vim#L27-L29)):

```
    ...
    if a:bang
        let &filetype = len(inputs) > 1 ? inputs[0] : a:user_input
    endif
    ...
```

Knowing that, I went on and re-set it back to its original value only when `neoformat#Neoformat()` was invoked with a bang and when the filetype was actually changed.

Let me know if there is something I am missing.

Reply:
Oh right, makes sense!